### ISSUE TITLE
`auto canary` doesn't seem to work with lerna independent mode

### ISSUE THREAD

**Issue by egoist** [intuit/auto#370]
**Describe the bug**

Auto throws in Lerna independent mode:

```
❯ yarn auto canary --pr 137
yarn run v1.13.0
$ /Users/egoist/dev/saber/node_modules/.bin/auto canary --pr 137
TypeError: Invalid Version: vindependent
    at new SemVer (/Users/egoist/dev/saber/node_modules/auto/node_modules/semver/semver.js:323:11)
    at compare (/Users/egoist/dev/saber/node_modules/auto/node_modules/semver/semver.js:614:10)
    at Function.gt (/Users/egoist/dev/saber/node_modules/auto/node_modules/semver/semver.js:643:10)
    at /Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:121:52
    at step (/Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:32:23)
    at Object.next (/Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:13:53)
    at fulfilled (/Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:4:58)
    at processTicksAndRejections (internal/process/task_queues.js:88:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**To Reproduce**

1. In a repo using Lerna independent mode
2. Run `auto canary` against a PR

**Expected behavior**

`auto canary` should publish a pre release

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. iOS] macOS

**Additional context**

Auto version: v4.15.5

**Comment by zephraph** [comment]
Hmmmm... I don't think auto is setup to handle independent mode at all. That'd be tricky cause you'd need to know which package(s) were updated.

**Comment by egoist** [comment]
so a better error message would be nice 😄 maybe mention this in the docs as well.

**Comment by zephraph** [comment]
Fair point. 

What behavior would you expect from running this?

**Comment by hipstersmoothie** [comment]
I am exploring independent mode now. I agree we should have a good error message for that, I'll make a PR.

There are a few tricky things about it but the hardest thing to get around is a PR that effect multiple packages. When lerna is used with conventional commits it is easy to say on package gets a major, minor, or patch since you know what commits contain what type of changes. In auto it only knows how the PR affects the semver. 

If that isn't a requirement of independent mode I think we can maybe support something. But it's a hard maybe, might require a bit of restructuring

**Comment by hipstersmoothie** [comment]
Hmm another hurdle:

A lot of auto relies on the ability to get the the "Latest Release" from github. In independent mode there really is not a latest release. Well at least the latest release number has no bearing. Might be able to add a hook to get around that

**Comment by hipstersmoothie** [comment]
I think it should be possible 

- Canary
    - Need to move version calculation to hook and call it with the bump 
- Changeling
    - Add title creation hook so no vnull. maybe just use root package name
    - Could put version on changlog lines instead
- Release
    - Make line where we getCurrentVersion a hook. 
    - Then the npm package can say what Independent tag is the latest. 
    - Move the prefixed release line into plugin too

**Comment by hipstersmoothie** [comment]
I took an initial stab at it. You can try it out using version `5.1.0-canary.373.4594`

**Comment by adierkens** [comment]
:rocket: Issue was released in v6.0.0 :rocket:
